### PR TITLE
Feature/sc 9201 student homework dashboard panels

### PR DESCRIPTION
**Pull request by martynakrysinska**
# Short Description

This PR includes:
- refactoring of HomeworksDashboardPanels component
- integrating HomeworksDashboardPanels component into HomeworksDashboardStudent component
- some code clean up of the tasks dashboards 

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,12 @@ Allowed Types of change: `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `
 - SC-9158 - fixed launch config to debug a single test
 - SC-9160 - fixed getStatus getter in service-template
 
+## [26.5.2] - 2021-07-12
+
+### Added
+
+- SC-9189 - Adding expansion panels to teacher's task dashboard
+
```
Comment (dominikbruechner) [inline CHANGELOG.md:42]: I guess you could move that to unreleased

Comment (dominikbruechner) [reply CHANGELOG.md:42]:
```suggestion
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "schulcloud-nuxt",
-	"version": "26.5.1",
+	"version": "26.5.2",
```
Comment (dominikbruechner) [inline package.json:3]:
```suggestion
	"version": "26.5.1",
```
probably we can leave that for now at 26.5.1

File: src/components/organisms/HomeworksDashboardMain.stories.js [hunk 3]
```diff
@@ -31,7 +31,6 @@ storiesOf("0 Vuetify/Homeworks/Templates", module)
 						isListEmpty: () => false,
 						isListFilled: () => true,
 						getCourses: () => coursesStudent,
-						getOpenHomeworks: () => homeworks,
```
Comment (dominikbruechner) [inline src/components/organisms/HomeworksDashboardMain.stories.js:34]: why does this need to go? all good if anticipated

File: package-lock.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "schulcloud-nuxt",
-	"version": "26.5.1",
+	"version": "26.5.2",
```
Comment (dominikbruechner) [inline package-lock.json:3]:
```suggestion
	"version": "26.5.1",
```

File: src/components/organisms/HomeworksDashboardPanels.vue [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+				</v-expansion-panel-content>
+			</v-expansion-panel>
+			<v-expansion-panel :disabled="isPanelTwoDisabled">
+				<v-expansion-panel-header
+					v-if="isListFilled"
+					class="text-h6 font-weight-bold pa-0"
+				>
+					{{ $t("pages.homeworks.teacher.subtitleWithDue") }}
+					<template v-slot:actions>
+						{{ dueDateHomeworks.length + overDueHomeworks.length }}
```
Comment (dominikbruechner) [inline src/components/organisms/HomeworksDashboardPanels.vue:32]: here I fixed a small bug, you missed the overDueHomeworks

File: src/components/organisms/HomeworksDashboardPanels.vue [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+		};
+	},
+	computed: {
+		...mapGetters("homeworks", {
+			dueDateHomeworks: "getOpenHomeworksWithDueDate",
+			overDueHomeworks: "getOverDueHomeworks",
+			noDueDateHomeworks: "getOpenHomeworksWithoutDueDate",
+			isListFilled: "isListFilled",
+			status: "getStatus",
+		}),
```
Comment (dominikbruechner) [inline src/components/organisms/HomeworksDashboardPanels.vue:65]: Now I understand what you meant on rc.

I am not totally agreeing with this approach, as of the following reasoning:
- The dashboardPanel component should, in my opinion, be dumb and shouldn't care about slot content and data structure
- By mapping the getters you basically make it resistant to reuse
- I would even go further and define even the headlines as props, and then rename the component to vDoublePanels or something

You can also leave it as is, taking this into account, and adapt it when needed at some point.

File: src/components/organisms/HomeworksDashboardTeacher.unit.js [hunk 7]
```diff
@@ -52,7 +52,7 @@ describe("@components/organisms/HomeworksDashboardTeacher", () => {
 
 	it(...isValidComponent(HomeworksDashboardTeacher));
 
-	it("Should render homeworks list component, with both panels expanded per default", () => {
+	it("Should render homeworks list component, with second panel expanded per default", () => {
```
Comment (dominikbruechner) [inline src/components/organisms/HomeworksDashboardTeacher.unit.js:55]: thanks 👏

File: src/components/organisms/HomeworksList.vue [hunk 8]
```diff
@@ -65,9 +65,7 @@ export default {
 		...mapGetters("homeworks", {
 			status: "getStatus",
 		}),
-	},
-	methods: {
-		isListFilled() {
+		isListFilled: function () {
```
Comment (dominikbruechner) [inline src/components/organisms/HomeworksList.vue:68]: 👏

**Review by dominikbruechner — CHANGES_REQUESTED** [review]
Looks good, besides what I wrote for the DashboardPanels component and the following:

The teacher's unit tests include:
- it("Should render not disabled 'no due date' panel, and disabled empty other panel", 
- it("Should render not disabled 'with due date' panel, and disabled empty other panel", 

these two tests can be generalized, and then unit test the HomeworksDashboardPanels component directly. 

otherwise todo bien nice work 🥇

File: src/components/organisms/HomeworksDashboardMain.stories.js (hunk 3 above)
Comment (martynakrysinska) [reply src/components/organisms/HomeworksDashboardMain.stories.js:34]: we're not using this getter now when the student also has homeworks without due date listed separately

File: src/components/organisms/HomeworksDashboardPanels.vue (hunk 5 above)
Comment (martynakrysinska) [reply src/components/organisms/HomeworksDashboardPanels.vue]: thanks!

File: src/components/organisms/HomeworksDashboardMain.stories.js (hunk 3 above)
Comment (martynakrysinska) [reply src/components/organisms/HomeworksDashboardMain.stories.js:34]: thanks, tomorrow i will still work on the tests, hence WIP label

File: src/components/organisms/HomeworksDashboardPanels.unit.js [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+	it("Should disable each panel, if its count is 0", () => {
+		const wrapper = mount(HomeworksDashboardPanels, {
+			...createComponentMocks({
+				i18n: true,
+				vuetify: true,
+			}),
+			vuetify,
+			propsData: propsDataOneEmpty,
+		});
+		expect(wrapper.find(".v-expansion-panel--disabled").exists()).toBe(true);
```
Comment (dominikbruechner) [inline src/components/organisms/HomeworksDashboardPanels.unit.js:110]:
```suggestion
		const expansionPanels = wrapper.findAll(".v-expansion-panel");
		expect(expansionPanels.exists()).toBe(true);
		expect(expansionPanels.at(0).classes()).toContain(
			"v-expansion-panel--disabled"
		);
		expect(expansionPanels.at(1).classes()).toContain(
			"v-expansion-panel--disabled"
		);
```

File: src/components/organisms/HomeworksDashboardPanels.vue [hunk 10]
```diff
@@ -0,0 +1,112 @@
+<template>
```
Comment (dominikbruechner) [inline src/components/organisms/HomeworksDashboardPanels.vue:1]: we need a storybook entry as well for this component (for the screenshot tests)

**Comment by dominikbruechner** [comment]
@schul-cloud-bot update screenshots

**Comment by martynakrysinska** [comment]
@schul-cloud-bot update screenshots

**Review by dominikbruechner — APPROVED** [review]

**Pull request merged**